internal/grpc/services/gateway/usershareprovider.go

```diff
@@ -252,6 +316,71 @@ func (s *svc) ListReceivedShares(ctx context.Context, req *collaboration.ListRec
 	return res, nil
 }
 
+func (s *svc) ListExistingReceivedShares(ctx context.Context, req *collaboration.ListReceivedSharesRequest) (*gateway.ListExistingReceivedSharesResponse, error) {
+	rshares, err := s.ListReceivedShares(ctx, req)
+	if err != nil {
+		err := errors.Wrap(err, "gateway: error calling ListExistingReceivedShares")
+		return &gateway.ListExistingReceivedSharesResponse{
+			Status: status.NewInternal(ctx, err, "error listing received shares"),
+		}, nil
+	}
+
+	sharesCh := make(chan *gateway.ReceivedShareResourceInfo, len(rshares.Shares))
+	pool := pond.NewPool(50)
+	for _, rs := range rshares.Shares {
+		rs := rs
+		pool.SubmitErr(func() error {
+			if rs.State == collaboration.ShareState_SHARE_STATE_REJECTED || rs.State == collaboration.ShareState_SHARE_STATE_INVALID {
```

Comment:
I think we use `collaboration.ShareState_SHARE_STATE_REJECTED` for `hidden`, so I don't think we should return an error for this?